### PR TITLE
Tidy iputils.hh and iputils.cc

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Especially the "extra tricky" commit needs extra review attention.

Not terribly happy about the NOLINT markers, but no way to work around that without a major overhaul, and even then  not all of them can be avoided, as we are calling into APIs that use old-school C idiom.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/iputils.cc [hunk 1]
```diff
@@ -323,61 +327,67 @@ int sendOnNBSocket(int fd, const struct msghdr *msgh)
...
+size_t sendMsgWithOptions(int fileDesc, const char* buffer, size_t len, const ComboAddress* dest, const ComboAddress* local, unsigned int localItf, int flags)
 {
-  struct msghdr msgh;
-  struct iovec iov;
+  struct msghdr msgh
+  {
+  };
+  struct iovec iov
+  {
+  };
```
Comment (rgacogne) [inline pdns/iputils.cc:382]:
```suggestion
  msghdr msgh{};
  iovec iov{};
```

File: pdns/iputils.hh [hunk 2]
```diff
@@ -83,39 +83,44 @@
 #undef IP_PKTINFO
 #endif
 
-union ComboAddress {
-  struct sockaddr_in sin4;
+union ComboAddress
+{
+  struct sockaddr_in sin4
+  {
+  };
```
Comment (rgacogne) [inline pdns/iputils.hh:90]:
```suggestion
  sockaddr_in sin4{};
```

**Review by rgacogne — APPROVED** [review]

File: pdns/iputils.cc (hunk 1 above)
Comment (omoerbeek) [reply pdns/iputils.cc:382]: clang-format insists, unless I remove `struct`.

Comment (rgacogne) [reply pdns/iputils.cc]: Yes, that's why I'm suggesting to remove `struct` :)

Comment (omoerbeek) [reply pdns/iputils.cc]: Ah, did not spot that in your suggestion :-(

**Review by rgacogne — APPROVED** [review]

**Pull request merged**